### ISSUE TITLE
Fix bug in AJAX smart search with Cyrillic words

### ISSUE THREAD

**Issue by Avalon18** [joomla/joomla-cms#31532]
in Joomla version 3.9.22, AJAX smart search suggestions worked, but after updating to Joomla 3.9.23, AJAX stopped working.
problem in the file /public_html/components/com_finder/models/suggestions.php from the Joomla 3.9.23 update

In the file components\com_finder\models\suggestions.php changing to

```
// connecting the class after at the beginning of the file defined('_JEXEC') or die;
use Joomla\String\StringHelper;

//replacing the bug code with a good one
//$mappingTableSuffix = substr(md5(substr($this->getState('input'), 0, 1)), 0, 1);
$mappingTableSuffix = StringHelper::substr(md5(StringHelper::substr($this->getState('input'), 0, 1)), 0, 1);
```

**Comment by HLeithner** [comment]
Thank  @AlexeyGal to figure this out, can you create a pull request?

**Comment by Avalon18** [comment]
> Thank @AlexeyGal to figure this out, can you create a pull request?

Yes. I create pull request now

**Comment by bembelimen** [comment]
Closed as having a PR: #31533
